internal_transfers/actions/src/parse.ts

```diff
@@ -0,0 +1,55 @@
+import { SettlementEvent, TradeEvent, TransferEvent } from "./models";
+import { Log } from "@tenderly/actions";
+import { addressFromBytes32, transferInvolves } from "./utils";
+
+export const SETTLEMENT_CONTRACT_ADDRESS =
+  "0x9008d19f58aabd9ed0d60971565aa8510560ab41";
+export const SETTLEMENT_EVENT_TOPIC =
+  "0x40338ce1a7c49204f0099533b1e9a7ee0a3d261f84974ab7af36105b8c4e9db4";
+export const TRANSFER_EVENT_TOPIC =
+  "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef";
+export const TRADE_EVENT_TOPIC =
+  "0xa07a543ab8a018198e99ca0184c93fe9050a79400a0a723441f84de1d972cc17";
+
+export interface ClassifiedEvents {
+  trades: TradeEvent[];
+  transfers: TransferEvent[];
+  settlements: SettlementEvent[];
+}
+
+export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
+  let result: ClassifiedEvents = {
+    transfers: [],
+    trades: [],
+    settlements: [],
+  };
+  logs.forEach((log, index) => {
+    const topics = log.topics;
```

Comment:
IMO It would be cleaner to assign each topic to a separate variable with a meaningful name instead of keeping them inside of the list.

Reply:
I suppose you are referring to the next line here then?

```
    const eventTopic = topics[0];
```

I think this is pretty common notation even on etherscan and there really isn't much of a better name here. They are literally called `index_topic_0`, `index_topic_1` and `index_topic_2` (like exactly how they appear in the array). I don't belive we will be manually parsing these event logs anymore soon (based on other feedback) so all the contents of this loop will likely wind up "under the hood".

![Screenshot 2023-02-13 at 13 48 00](https://user-images.githubusercontent.com/11778116/218461621-86f7ce08-dd89-473e-b647-f48fa5f8c862.png)

Reply:
Now that we have implemented https://github.com/cowprotocol/solver-rewards/pull/193 this is no longer an issue (since we do not manually parse raw transaction data anymore, but rather user a library for it).